CODESTYLE.md

```diff
@@ -0,0 +1,125 @@
+You can find an updated list of rules and more information about regarding all available rules in [Rules.md](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md).
+
+- [Swift Lint](https://github.com/realm/SwiftLint) to enforce Swift style and perfrorm some lint checks.
+
+All rules for `SwiftLint` are configured in `.swiftlint.yml` file. Additionally `SwiftLint` has some custom rules. 
+You can find an updated list of rules and more information about regarding all available rules in [Rules.md](https://github.com/realm/SwiftLint/blob/master/Rules.md).
+You can also check [Source/SwiftLintFramework/Rules directory](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules) to see their implementation.
+
+- [ClangFormat](https://clang.llvm.org/docs/ClangFormat.html) to handle formatting for Obj-C/Obj-C++ code (`.h`/`.m`/`.mm` file extensions).
+
+## Installation and usage
+
+Please note we're using `Swift 4.2` as a minimal version, thus make sure you install appropriate toolchain or switch to it in case you have more than one. Otherwise you might face some false positivis triggers or build errors.
+
+### Install Swiftlint
+
+Following options to install tools are available:
+
+1. Using Homebrew:
+
+```
+brew update
+brew install swiftlint
+```
+
+2. Using a pre-built package:
+
+Install SwiftLint by downloading `SwiftLint.pkg` from the [latest GitHub release](https://github.com/realm/SwiftLint/releases/) and running it.
+
+3. Compiling from source:
+
+You can also build from source by cloning [Swiftlint project](https://github.com/realm/SwiftLint) and running (Xcode 10.0 or later):
+```
+git submodule update --init --recursive; make install
+```
+
+### How to use Swiftlint:
+
+- Xcode
+
+Swiftlint is integrated into an Xcode build phase, so you'll get warnings and errors displayed in the IDE each time you hit `Cmd + B`  or `Cmd + R`.
+
+- AppCode
+
+To integrate SwiftLint with AppCode, install [this plugin](https://plugins.jetbrains.com/plugin/9175) and configure SwiftLint's installed path in the plugin's preferences. The autocorrect action is available via `⌥⏎.`
+
+- Running manually as a command-line tool
+
+You can run swiftlint in the directory containing the Swift files to lint:
+```
+swiftlint
+```
+Directories will be searched recursively.
+
+### Install Swiftformat
+
+1. Using Homebrew:
+
+```
+brew update
+brew install swiftformat
+```
+
+2. Compiling from source:
+
+If you prefer, you can check out and build SwiftFormat manually on macOS or Linux as follows:
+```
+git clone https://github.com/nicklockwood/SwiftFormat   
+cd SwiftFormat
+swift build -c release
+```
+You can find more details at [Swiftformat page](https://github.com/nicklockwood/SwiftFormat#command-line-tool)
+
+### How to use Swiftformat:
+
+- Xcode
+
+Swiftlint is integrated into an Xcode build phase, so you'll get warnings displayed in the IDE each time you hit `Cmd + B`  or `Cmd + R`.
```

Comment:
@dersim-davaod shouldn't be here

Reply:
@IodaMikeMapbox, I'm sorry, could you please explain what's the issue? Should we reformulate `Swiftlint is integrated` part or you propose to remove this section at all?

Reply:
@dersim-davaod it's not related to `how to use swiftformat`, it's duplication from swiftlint's xcode section (line 74)

Reply:
Good catch, thanks! PTAL at new version.